lib/matplotlib/quiver.py

```diff
@@ -520,11 +532,9 @@ def get_datalim(self, transData):
     @allow_rasterization
     def draw(self, renderer):
         self._init()
-        if (self._new_UV or self.angles == 'xy'
-                or self.scale_units in ['x', 'y', 'xy']):
-            verts = self._make_verts(self.U, self.V)
-            self.set_verts(verts, closed=False)
-            self._new_UV = False
+        verts = self._make_verts(self.U, self.V)
```

Comment:
Yes,  exactly.  Originally I was trying to gain a little efficiency when possible.  The condition was already overly complicated.  Including one more criterion based on orientation would make it even worse, for no real gain.